File: gdx2d-desktop/src/hevs/gdx2d/demos/image_drawing/DemoMirrorImage.java
package hevs.gdx2d.demos.image_drawing;

import hevs.gdx2d.components.bitmaps.BitmapImage;
import hevs.gdx2d.lib.GdxGraphics;
import hevs.gdx2d.lib.PortableApplication;

/**
 * This demo is similar to {@link DemoSimpleImage} but demonstrates
 * how to use the mirrorUpDown and mirrorLeftRight methods
 *
 * @author Pierre-André Mudry (mui)
 * @version 1.0
 */
public class DemoMirrorImage extends PortableApplication {
	float counter = 0;
	BitmapImage imgBitmap;

	/**
	 * Constructor
	 *
	 * @param onAndroid tells if we are currently running on Android or not
	 */
	public DemoMirrorImage(boolean onAndroid) {
		super(onAndroid);
	}

	public static void main(String[] args) {
		new DemoMirrorImage(false);
	}

	@Override
	public void onInit() {
		// Sets the window title
		setTitle("Mirror image drawing, mui 2013");

		// Loads the image that will be displayed in the middle of the screen
		imgBitmap = new BitmapImage("data/images/Android_PI_48x48.png");
	}

	@Override
	public void onGraphicRender(GdxGraphics g) {
		/**
		 * Rendering
		 */
		// Clear the screen
		g.clear();

		// Render an with mirror
		g.drawPicture(getWindowWidth() / 3.0f, getWindowHeight() / 3.0f, imgBitmap);
		imgBitmap.mirrorUpDown();
		g.drawPicture(getWindowWidth() * 2.0f / 3.0f, getWindowHeight() / 3.0f, imgBitmap);
		imgBitmap.mirrorUpDown();

		g.drawPicture(getWindowWidth() / 3.0f, getWindowHeight() * 2.0f / 3.0f, imgBitmap);
		imgBitmap.mirrorLeftRight();
		g.drawPicture(getWindowWidth() * 2.0f / 3.0f, getWindowHeight() * 2.0f / 3.0f, imgBitmap);
		imgBitmap.mirrorLeftRight();

		g.drawFPS();        // Draws the number of frame per second
		g.drawSchoolLogo(); // Draws the school logo
	}
}

File: gdx2d-demoDesktop/src/main/scala/ch/hevs/gdx2d/demos/image_drawing/DemoMirrorImage.scala
package ch.hevs.gdx2d.demos.image_drawing

import ch.hevs.gdx2d.components.bitmaps.BitmapImage
import ch.hevs.gdx2d.desktop.DesktopApplication
import ch.hevs.gdx2d.lib.GdxGraphics

/**
 * Similar to [[DemoSimpleImage]] but demonstrates how to use the
 * `mirrorUpDown` and `mirrorLeftRight` methods.
 *
 * @author Pierre-André Mudry (mui)
 * @version 2.0
 */
class DemoMirrorImage extends DesktopApplication {

  private var imgBitmap: BitmapImage = _

  override def onInit(): Unit = {
    setTitle("Mirror image drawing, mui 2013")
    imgBitmap = new BitmapImage("images/Android_PI_48x48.png")
  }

  override def onGraphicRender(g: GdxGraphics): Unit = {
    g.clear()

    g.drawPicture(getWindowWidth / 3.0f, getWindowHeight / 3.0f, imgBitmap)
    imgBitmap.mirrorUpDown()
    g.drawPicture(getWindowWidth * 2.0f / 3.0f, getWindowHeight / 3.0f, imgBitmap)
    imgBitmap.mirrorUpDown()

    g.drawPicture(getWindowWidth / 3.0f, getWindowHeight * 2.0f / 3.0f, imgBitmap)
    imgBitmap.mirrorLeftRight()
    g.drawPicture(getWindowWidth * 2.0f / 3.0f, getWindowHeight * 2.0f / 3.0f, imgBitmap)
    imgBitmap.mirrorLeftRight()

    g.drawFPS()
    g.drawSchoolLogo()
  }
}

object DemoMirrorImage {
  def main(args: Array[String]): Unit = new DemoMirrorImage().launch()
}
